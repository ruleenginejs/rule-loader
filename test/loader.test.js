const compiler = require('./compiler.js');

test('compile simple rule', async () => {
  const stats = await compiler('simple.rule');
  const output = stats.toJson({ source: true }).modules[0].source;
  expect(output).toMatchSnapshot();
});

test('bad schema rule', async () => {
  expect.assertions(1);
  let message = null;
  try {
    await compiler('bad-schema.rule');
  } catch (e) {
    message = e[0].message.split("\n")[1];
  }
  expect(message).toMatchSnapshot();
});
