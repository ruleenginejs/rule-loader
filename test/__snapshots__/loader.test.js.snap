// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule-loader should output compiled rule module with commonjs module 1`] = `
"const {
  Pipeline,
  StartStep,
  SingleStep,
  EndStep
} = require(\\"@ruleenginejs/runtime\\");

const pipeline = new Pipeline();
const startStep_1 = new StartStep({
  id: 2,
  name: \\"Start\\",
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
const singleStep_2 = new SingleStep({
  id: 5,
  name: \\"New Step\\",
  handler: require(\\"./new-step.js\\"),
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
const endStep_3 = new EndStep({
  id: 8,
  name: \\"End\\",
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
startStep_1.connectTo(singleStep_2, \\"default\\", \\"default\\");
singleStep_2.connectTo(endStep_3, \\"default\\", \\"default\\");
pipeline.add(startStep_1, singleStep_2, endStep_3);
module.exports = pipeline;"
`;

exports[`rule-loader should output compiled rule module with es module 1`] = `
"import { Pipeline, StartStep, SingleStep, EndStep } from \\"@ruleenginejs/runtime\\";
import handler_3 from \\"./new-step.js\\";
const pipeline = new Pipeline();
const startStep_1 = new StartStep({
  id: 2,
  name: \\"Start\\",
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
const singleStep_2 = new SingleStep({
  id: 5,
  name: \\"New Step\\",
  handler: handler_3,
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
const endStep_4 = new EndStep({
  id: 8,
  name: \\"End\\",
  ports: {
    in: [\\"default\\"],
    out: [\\"default\\"]
  }
});
startStep_1.connectTo(singleStep_2, \\"default\\", \\"default\\");
singleStep_2.connectTo(endStep_4, \\"default\\", \\"default\\");
pipeline.add(startStep_1, singleStep_2, endStep_4);
export default pipeline;"
`;

exports[`rule-loader should throw error when invalid rule schema 1`] = `"Error: Schema invalid: [{\\"instancePath\\":\\"\\",\\"schemaPath\\":\\"#/required\\",\\"keyword\\":\\"required\\",\\"params\\":{\\"missingProperty\\":\\"steps\\"},\\"message\\":\\"must have required property 'steps'\\"}]"`;
